### PR TITLE
Add IfcGroundReinforcmentElement

### LINKED ISSUE #385: [IFC Tunnel] Add IfcGroundReinforcementElement
Including PDTs:
- SPILINGBOLT : Definition
- ROCKSUPPORTBOLT : Definition

See #367 and #365

### PR DESCRIPTION
**Pull request by larswik**
Fixes #385.
Check definitions!

### CODE REVIEW

File: IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/Entities/IfcGroundReinforcementElement/Documentation.md [hunk 1]
```diff
@@ -0,0 +1 @@
+An element that serves as reinforcement of the existing ground
```
Comment (SergejMuhic) [inline IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/Entities/IfcGroundReinforcementElement/Documentation.md]:
```suggestion
An element that serves as reinforcement of the existing ground.
```

File: IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/Entities/IfcGroundReinforcementElement/DocEntity.xml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+<DocEntity xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="IfcGroundReinforcementElement" UniqueId="eff9d345-72ca-4b29-8857-cc24f5c55fb0" BaseDefinition="IfcBuiltElement" EntityFlags="32">
+	<Attributes>
+		<DocAttribute Name="PredefinedType" UniqueId="5bf261d3-adfb-4e71-89d3-59707720d1ca" DefinedType="IfcGroundReinforcementElementTypeEnum">
+			<Documentation>Identifies the predefined type of ground reinforcementl elements. This type may associate additional specific property sets.</Documentation>
+		</DocAttribute>
+	</Attributes>
+	<WhereRules>
+		<DocWhereRule Name="CorrectPredefinedType" UniqueId="566bf0f8-aaeb-41ca-8145-127664b1dbee">
+			<Documentation>Either the _PredefinedType_ attribute is unset, or the inherited attribute _ObjectType_ shall be provided, if the _PredefinedType_ is set to USERDEFINED.</Documentation>
+			<Expression>NOT(EXISTS(PredefinedType)) OR
```
Comment (SergejMuhic) [inline IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/Entities/IfcGroundReinforcementElement/DocEntity.xml]: Probably should use HasObjectType rule. PredefinedType looks mandatory already.

**Review by SergejMuhic — CHANGES_REQUESTED** [review]

File: IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/Entities/IfcGroundReinforcementElement/DocEntity.xml (hunk 2 above)
Comment (larswik) [reply IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/Entities/IfcGroundReinforcementElement/DocEntity.xml]: Should be fixed now.

File: IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/Entities/IfcGroundReinforcementElement/DocEntity.xml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocEntity xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" Name="IfcGroundReinforcementElement" UniqueId="eff9d345-72ca-4b29-8857-cc24f5c55fb0" BaseDefinition="IfcBuiltElement" EntityFlags="32">
+	<Attributes>
+		<DocAttribute Name="PredefinedType" UniqueId="5bf261d3-adfb-4e71-89d3-59707720d1ca" DefinedType="IfcGroundReinforcementElementTypeEnum">
+			<Documentation>Identifies the predefined type of ground reinforcementl elements. This type may associate additional specific property sets.</Documentation>
```
Comment (anborr) [inline IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/Entities/IfcGroundReinforcementElement/DocEntity.xml]:
```suggestion
			<Documentation>Identifies the predefined type of ground reinforcement elements. This type may associate additional specific property sets.</Documentation>
```

**Review by anborr — CHANGES_REQUESTED** [review]

File: IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/Entities/IfcGroundReinforcementElement/DocEntity.xml (hunk 3 above)
Comment (SergejMuhic) [inline IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/Entities/IfcGroundReinforcementElement/DocEntity.xml]:
```suggestion
			<Documentation>Specifies the type for which the value is selected from a predefined type enumeration. This type may associate additional specific property sets.</Documentation>
```

Comment (SergejMuhic) [inline IFC4x3/Sections/Shared element data schemas/Schemas/IfcSharedInfrastructureElements/Entities/IfcGroundReinforcementElement/DocEntity.xml]:
```suggestion
			<Documentation>Specifies the type for which the value is selected from a predefined type enumeration. This type may associate additional specific property sets.</Documentation>
```

**Review by SergejMuhic — APPROVED** [review]

**Pull request merged**